wcomponents-theme/src/main/js/wc/file/util.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * File util module, to perform file operations like converting data to blob,
+ * blob to file, fixing file extension.
+ * This module defines are a set of mime type to extension mapping, the user could choose
+ * to override this map by 'wc/config' using id "wc/file/customMimeToExt".
+ * @module
+ */
+define(["wc/dom/uid", "wc/file/getMimeType", "wc/config"], function (uid, getMimeType, wcconfig) {
+	"use strict";
+
+	var instance = new FileUtil(),
+
+		/**
+		 * Map of mimetype to extension, used when checking the newly created file is named 
+		 * with the correct extension.
+		 **/
+		mimeToExt = {
+			"image/jpeg": ["jpeg", "jpg"],
+			"image/bmp": ["bmp"],
+			"image/png": ["png"],
+			"image/gif": ["gif"],
+			"image/x-icon": ["ico"],
+			"image/webp": ["webp"],
+			"image/svg+xml": ["svg"],
+			"image/tiff": ["tif", "tiff"],
+			"text/plain": ["txt", "bas", "c", "h", "log", "bat"],
+			"text/tab-separated-values": ["tsv"],
+			"text/richtext": ["rtx"],
+			"text/html": ["htm", "html", "stm"],
+			"text/xml": ["xml"],
+			"application/msword": ["doc", "dot"],
+			"application/vnd.ms-powerpoint": ["pot,", "pps", "ppt"],
+			"application/rtf": ["rtf"],
+			"application/pdf": ["pdf"],
+			"application/vnd.ms-excel": ["xla", "xlc", "xlm", "xls", "xlt", "xlw"]
```

Comment:
What extension would you generally expect an Excel file to be given? Would you expect "xla"?